scripts/check/tidy.sh

```
@@ -0,0 +1,55 @@
+    branch="$(git rev-parse --abbrev-ref HEAD)"
+    diff_base="$( [[ "${branch}" == "main" ]] && echo "HEAD^" || echo "origin/main" )"
+    files="$(git diff --name-only "${diff_base}" | grep '\.cpp$' | grep '^src/' || true)"
+
+    if [[ -z "${files}" ]]; then
+      echo "  [skip] no changed files vs ${diff_base}"
+    else
+      for dir in src $(find src -maxdepth 1 -mindepth 1 -type d); do
+        dir_files="$(echo "${files}" | grep "^${dir}/[^/]*\.cpp$" || true)"
+        if [[ -n "${dir_files}" ]]; then
+          echo "  [checking] ${dir}/ ($(echo "${dir_files}" | wc -w) files)"
+          ${CLANG_TIDY} --config-file=.config/.clang-tidy ${dir_files} -- -std=c++17 -I src/ 2>&1 \
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)